### ISSUE TITLE
blog does not pull in "not found" template (flow router)

### ISSUE THREAD

**Issue by deborah-ufw** [Differential/meteor-blog#272]
Blog presents empty content where "not found" content should be.
- using FlowRouter
- in FlowRouter routes, a default "not found" route is specified
- it works in the rest of the app
- blog is using a custom BlogPost.html
- the custom BlogPost.html is located in the same client/templates folder as the error-404.html template
- did not specify anything, but blog did not use the FlowRouter "not found" route
- tried specifying **blogNotFoundTemplate: 'error-404',**  in blog-config.js but it didn't work

Tried changing the template name from "error-404" to "error404", adjusting references and re-testing, but it still did not work.

**Comment by deborah-ufw** [comment]
When I changed my "error-404" template to "error404", it did work. The dash was the problem, although other templates have dashes and load all right.  Is this a bug in blog?

**Comment by foobarbecue** [comment]
I've been trying to understand meteor-blog's router recently (in router.coffee). Would be great if there's a dev here who can explain it.

It seems it's not really using flow-router or iron-router for the actual routing. I'm using flow-router, and meteor-blog registers a single route for the url pattern /:any\* and then seems to do the actual routing "manually." Unless I'm misunderstanding something, this seems like a very bad way of doing things and it's probably the reason that 404s are mishandled. Can anybody confirm or deny?

**Comment by hellogerard** [comment]
@foobarbecue This is true. This was the only way to support both routers at the same time.
